packages/flame/lib/src/collision_detection/hitboxes/hitbox.dart

```diff
@@ -0,0 +1,30 @@
+import '../../../collision_detection.dart';
+import '../../../extensions.dart';
+
+/// The [Hitbox] is the default building block to determine whether two objects
+/// have collided with each other. [HitboxShape] is the default implementation
+/// used in FCS.
+mixin Hitbox<T extends Hitbox<T>> implements GenericCollisionCallbacks<T> {
+  /// Whether the hitbox should:
+  ///   * [CollidableType.active] - actively collide with other hitboxes.
+  ///   * [CollidableType.passive] - passively collide with other hitboxes (only
+  ///   collide with hitboxes that are active, but not other passive ones).
+  ///   * [CollidableType.inactive] - not collide with any other hitboxes.
+  CollidableType get collidableType;
+
+  /// The axis-aligned bounding box of the [Hitbox], this is used to make a
+  /// rough estimation of whether two hitboxes can possibly collide or not.
+  Aabb2 get aabb;
+
+  /// Checks whether the [Hitbox] contains the [point].
+  bool containsPoint(Vector2 point);
```

Comment:
That is the point, since then it can be used in FCS directly by using this mixin on a component, but it can at the same time be used outside of FCS knowing that the `containsPoint` method will exist.

Reply:
The `point` is under-specified here: it doesn't say what the coordinate system is. Is it in local coordinates, parent's coordinates, world coordinates, canvas coordinates, device coordinates, or something else? The real danger then comes from when `Component` uses one coordinate system but the `Hitbox` mixin uses another. Remember how we had a trouble with `update()` in `Component`/`Game`?

Reply:
It is in the global coordinate system for `HitboxShape`, just like it is for `PositionComponent`, I'll make sure to add that in the dartdocs for `HitboxShape` too.
The `Hitbox` is general enough to not have to care about the coordinate system, that is left up to the implementer of the new type of hitbox.